llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -48,6 +48,9 @@ extern cl::opt<unsigned> ForceTargetInstructionCost;
 
 bool VPRecipeBase::mayWriteToMemory() const {
   switch (getVPDefID()) {
+  case VPInstructionSC: {
+    return !Instruction::isBinaryOp(cast<VPInstruction>(this)->getOpcode());
+  }
```

Comment:
What about cast or select or other instructions here? Is this still correct for them?

Reply:
Yep, unified the logic separately (68ed1728bf45162187a2b54eed226097b8fc0a12) and remove the change here